File: experiments/runs/2026-09-20_06-41-41_claim-office-java-example-mapping_exact-ptdd-v1-pi_gpt-5-6-sol-codex/src/main/java/Policy.java
import com.fasterxml.jackson.databind.JsonNode;

final class Policy {
    private final JsonNode items;
    private int remainingCap;

    Policy(JsonNode items, int insuranceSum) {
        this.items = items.deepCopy();
        this.remainingCap = insuranceSum * 2;
    }

    JsonNode items() {
        return items;
    }

    int remainingCap() {
        return remainingCap;
    }

    int payUpToCap(int desiredPayout) {
        int payout = Math.min(desiredPayout, remainingCap);
        remainingCap -= payout;
        return payout;
    }
}

File: experiments/runs/2026-09-23_16-45-13_claim-office-example-mapping_exact-ptdd-v1.1-refactor-subagent-cc_opus-5-5-no-thinking/src/policy.ts
import type { Damage, Item } from "./claimOffice.js";
import { coveredDamages } from "./coverage.js";
import { rejectNegativeDamages } from "./damageReport.js";
import * as priceList from "./priceList.js";
import { damagePayout } from "./reimbursement.js";
import { roundPayoutInMHPCOsFavor } from "./rounding.js";

const CAP_MULTIPLIER = 2;

export interface Policy {
  items: Item[];
  remainingCap: number;
}

function insuranceSum(items: Item[]): number {
  return items.reduce((sum, item) => sum + priceList.insuranceValue(item.type), 0);
}

function payoutCap(items: Item[]): number {
  return CAP_MULTIPLIER * insuranceSum(items);
}

export function openPolicy(items: Item[]): Policy {
  return { items, remainingCap: payoutCap(items) };
}

function claimPayout(policy: Policy, damages: Damage[]): number {
  return coveredDamages(policy.items, damages).reduce(
    (sum, damage) => sum + damagePayout(damage.insuredItem, damage.amount),
    0,
  );
}

export interface ClaimResult {
  payout: number;
  remainingCap: number;
}

function drawFromRemainingCap(policy: Policy, desiredPayout: number): number {
  const payout = Math.min(policy.remainingCap, desiredPayout);
  policy.remainingCap -= payout;
  return payout;
}

export function processClaim(policy: Policy, damages: Damage[]): ClaimResult {
  rejectNegativeDamages(damages);
  const desiredPayout = roundPayoutInMHPCOsFavor(claimPayout(policy, damages));
  const payout = drawFromRemainingCap(policy, desiredPayout);
  return { payout, remainingCap: policy.remainingCap };
}
